mobile/lib/blocs/following/following_bloc.dart

```diff
@@ -0,0 +1,212 @@
+    // Listen to repository stream for reactive updates
+    await emit.forEach<List<String>>(
+      _followRepository.followingStream,
+      onData: (followingPubkeys) => state.copyWith(
+        status: FollowingStatus.success,
+        followingPubkeys: followingPubkeys,
+      ),
+      onError: (error, stackTrace) {
+        Log.error(
+          'Error in following stream: $error',
+          name: 'FollowingBloc',
+          category: LogCategory.system,
+        );
+        return state.copyWith(status: FollowingStatus.failure);
+      },
+    );
+  }
+
+  /// Load other user's following from Nostr relays
+  Future<void> _loadOtherUserFollowing(Emitter<FollowingState> emit) async {
+    // Cancel any existing subscription
+    await _nostrSubscription?.cancel();
+
+    final completer = Completer<List<String>>();
+
+    // Subscribe to the user's kind 3 contact list events
+    final eventStream = _nostrClient.subscribe([
+      Filter(
+        authors: [_targetPubkey],
+        kinds: const [3], // Contact lists
+        limit: 1, // Get most recent only
+      ),
+    ]);
+
+    // Apply timeout
+    final timeoutStream = eventStream.timeout(
+      const Duration(seconds: 5),
+      onTimeout: (sink) {
+        if (!completer.isCompleted) {
+          completer.complete([]);
+        }
+        sink.close();
+      },
+    );
+
+    _nostrSubscription = timeoutStream.listen(
+      (event) {
+        // Extract followed pubkeys from 'p' tags
+        final following = <String>[];
+        for (final tag in event.tags) {
+          if (tag.isNotEmpty && tag[0] == 'p' && tag.length > 1) {
+            final followedPubkey = tag[1];
+            if (!following.contains(followedPubkey)) {
+              following.add(followedPubkey);
+            }
+          }
+        }
+
+        if (!completer.isCompleted) {
+          completer.complete(following);
+        }
+      },
+      onError: (Object error) {
+        Log.error(
+          'Error fetching following list: $error',
+          name: 'FollowingBloc',
+          category: LogCategory.relay,
+        );
+        if (!completer.isCompleted) {
+          completer.completeError(error);
+        }
+      },
+      onDone: () {
+        if (!completer.isCompleted) {
+          completer.complete([]);
+        }
+      },
+    );
+
+    final following = await completer.future;
+    emit(
+      state.copyWith(
+        status: FollowingStatus.success,
+        followingPubkeys: following,
+      ),
+    );
+  }
+
+  /// Handle follow toggle request.
+  /// Determines whether to follow or unfollow based on current state. UI will consume the stream for inmediate update.
+  Future<void> _onFollowToggleRequested(
+    FollowToggleRequested event,
+    Emitter<FollowingState> emit,
+  ) async {
+    final isCurrentlyFollowing = _followRepository.isFollowing(event.pubkey);
+
+    try {
+      if (isCurrentlyFollowing) {
+        await _followRepository.unfollow(event.pubkey);
```

Comment:
nit: since i see this also on the other bloc wdyt having switch method, that will do the check if it should do follow and unfollow

Reply:
Good catch, and i think this fits better in the repository instead